src/vs/editor/contrib/hover/browser/markdownHoverParticipant.ts

```diff
@@ -124,17 +170,164 @@ export class MarkdownHoverParticipant implements IEditorHoverParticipant<Markdow
 			return AsyncIterableObject.EMPTY;
 		}
 
-		const position = new Position(anchor.range.startLineNumber, anchor.range.startColumn);
-		return getHover(this._languageFeaturesService.hoverProvider, model, position, token)
+		this._position = anchor.range.getStartPosition();
+		return getHover(this._languageFeaturesService.hoverProvider, model, this._position, undefined, token)
 			.filter(item => !isEmptyMarkdownString(item.hover.contents))
 			.map(item => {
 				const rng = item.hover.range ? Range.lift(item.hover.range) : anchor.range;
-				return new MarkdownHover(this, rng, item.hover.contents, false, item.ordinal);
+				const hoverVerbosity: HoverVerbosityMetadata = {
+					canIncreaseVerbosity: item.hover.canIncreaseVerbosity,
+					canDecreaseVerbosity: item.hover.canDecreaseVerbosity
+				};
+				return new VerboseMarkdownHover(this, rng, item.hover.contents, false, item.ordinal, item.hover.id, item.provider, hoverVerbosity, item.hover.dispose);
+			});
+	}
+
+	public renderHoverParts(context: IEditorHoverRenderContext, hoverParts: VerboseMarkdownHover[]): IDisposable {
+		this._context = context;
+		this._context.disposables?.add(toDisposable(() => {
+			this._hoverData.forEach(hover => {
+				hover.dispose?.();
 			});
+		}));
+		hoverParts.sort((a, b) => a.ordinal - b.ordinal);
+		this._hoverData = hoverParts.map(hover => { return { provider: hover.sourceProvider, id: hover.hoverId, dispose: hover.dispose }; });
```

Comment:
`dispose: hover.dispose` - this is dangerous and might not work! `hover.dispose` might lose `this`.
Why do need to map the items?

Reply:
I will directly save the hoverParts into a field without using a map